### PR TITLE
Add scalarThreshold to PipelineShaderOptions

### PR DESCRIPTION
**Pull request by Kai-AMD**

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit d170953c847b80970b6ce6a7cfbac4afa8c18466
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 831f334a9f4ef429b5d4e953b86b4624a7c2db54</li>
<li>SPVGEN: dev f1bc2ba988273c3724afffe72fe9cd933a022ce7</li>
<li>XGL: dev 54a1ba508fd82ed2a11d0960f0f2f35f975b2f12</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 0549a86140ab4e138db4f2a6ea72fbacb69bdddb</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 939/939 (100.0%)</li>
<li>Failed: 0/939 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 65433/65433 (100.0%)</li>
<li>Failed: 0/65433 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 65465/65465 (100.0%)</li>
<li>Failed: 0/65465 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 65465/65465 (100.0%)</li>
<li>Failed: 0/65465 (0.0%)</li>
</ul></details>


</ul></details>

File: include/llpc.h [hunk 1]
```diff
@@ -51,6 +51,7 @@
 //* %Version History
 //* | %Version | Change Description                                                                                    |
 //* | -------- | ----------------------------------------------------------------------------------------------------- |
+//* |     39.0 | Added scalarThreshold to PipelineShaderOptions                                                        |
```
Comment (amdrexu) [inline include/llpc.h]: I think the versioning is changed to just bump minor version (no vesion macro gating) without changing the major one if new fields are added to LLPC interface, not disrupting existing ones.

Comment (Kai-AMD) [reply include/llpc.h]: Already changed the version from 39.0 to 38.2 and remove all the version check as well.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 5eb0e1e4cd843c6109faed67e618a2bbb9b32edb
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 5d72cf9890fe2fbb1b87eecc497114e04afaa277</li>
<li>SPVGEN: dev f1bc2ba988273c3724afffe72fe9cd933a022ce7</li>
<li>XGL: dev 83c89cd727653c4514e2abe30e1633220fdb666e</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 0549a86140ab4e138db4f2a6ea72fbacb69bdddb</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 939/939 (100.0%)</li>
<li>Failed: 0/939 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 65433/65433 (100.0%)</li>
<li>Failed: 0/65433 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 65465/65465 (100.0%)</li>
<li>Failed: 0/65465 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 65465/65465 (100.0%)</li>
<li>Failed: 0/65465 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**